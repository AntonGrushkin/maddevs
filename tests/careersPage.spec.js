import {
  mount
} from '@vue/test-utils';
import CareersPage from '@/pages/_lang/careers';

describe('Careers page', () => {
  let wrapper;

  beforeEach(() => {
    global.$nuxt = {
      $route: {
        name: null
      }
    }
    wrapper = mount(CareersPage, {
      stubs: ['ValidationObserver']
    });
  });

  // ------ IMPORTANT ----- //
  test('is a Vue instance', () => {
    expect(wrapper.exists()).toBeTruthy();
  });
  
  test('renders correctly', () => {
    expect(wrapper.element).toMatchSnapshot();
  });
  // --------------------- //

  test('Sets the correctly data', () => {
    expect(wrapper.vm.$data.title).toEqual('Mad Devs Careers');
    expect(wrapper.vm.$data.description).toEqual('');
    expect(wrapper.vm.$data.ogUrl).toEqual('https://maddevs.io/careers');
  });
});