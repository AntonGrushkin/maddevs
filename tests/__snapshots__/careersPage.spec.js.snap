// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Careers page renders correctly 1`] = `
<main
  class="main"
>
  <section
    class="banner-container"
    id="banner-container"
  >
    <div
      class="container"
    >
      <!---->
       
      <!---->
       
      <!---->
       
      <!---->
    </div>
  </section>
   
  <section
    class="careers"
    id="careers"
  >
    <div
      class="container"
    >
      <div
        class="careers__wrapper"
      >
        <img
          alt="Careers Background Image"
          class="careers__background-logo"
          src="@/assets/img/Careers/svg/careers_logo.svg"
        />
         
        <validationobserver-stub />
      </div>
    </div>
     
    <transition-stub
      id="career-modal"
      name="fade"
    >
      <!---->
    </transition-stub>
  </section>
</main>
`;
